fixit/rules/use_assert_is_not_none.py

```diff
@@ -76,173 +86,86 @@ class UseAssertIsNotNoneRule(CstLintRule):
-                        )
-                    )
-                ],
-            ),
-        ):
-            new_call = node.with_changes(
-                func=cst.Attribute(
-                    value=cst.Name("self"), attr=cst.Name("assertIsNone")
-                ),
-                args=[cst.Arg(ensure_type(node.args[0].value, cst.Comparison).left)],
-            )
-            self.report(node, replacement=new_call)
+        if result:
+            assertion_name = result["assertion_name"]
+            if isinstance(assertion_name, Sequence):
+                assertion_name = assertion_name[0]
 
-        # `self.assertFalse(x is not None)` -> `self.assertIsNone(x)`
-        elif m.matches(
-            node,
-            m.Call(
-                func=m.Attribute(value=m.Name("self"), attr=m.Name("assertFalse")),
-                args=[
-                    m.Arg(
-                        m.Comparison(
-                            comparisons=[
-                                m.ComparisonTarget(m.IsNot(), comparator=m.Name("None"))
-                            ]
-                        )
-                    )
-                ],
-            ),
-        ):
-            new_call = node.with_changes(
-                func=cst.Attribute(
-                    value=cst.Name("self"), attr=cst.Name("assertIsNone")
-                ),
-                args=[cst.Arg(ensure_type(node.args[0].value, cst.Comparison).left)],
-            )
-            self.report(node, replacement=new_call)
-        # `self.assertFalse(not x is None)` -> `self.assertIsNone(x)`
-        elif m.matches(
-            node,
-            m.Call(
-                func=m.Attribute(value=m.Name("self"), attr=m.Name("assertFalse")),
-                args=[
-                    m.Arg(
-                        value=m.UnaryOperation(
-                            operator=m.Not(),
-                            expression=m.Comparison(
-                                comparisons=[
-                                    m.ComparisonTarget(
-                                        m.Is(), comparator=m.Name("None")
-                                    )
-                                ]
-                            ),
-                        )
-                    )
-                ],
-            ),
-        ):
+            argument = result["argument"]
+            if isinstance(argument, Sequence):
+                argument = argument[0]
+
+            comparison_type = result["comparison_type"]
+            if isinstance(comparison_type, Sequence):
+                comparison_type = comparison_type[0]
+
+            if m.matches(argument, m.Comparison()):
+                assertion_argument = ensure_type(argument, cst.Comparison).left
+            else:
+                assertion_argument = ensure_type(
+                    ensure_type(argument, cst.UnaryOperation).expression, cst.Comparison
+                ).left
 
+            negations_seen = 0
```

Comment:
The linter didn't like my first attempt with the many nested branches however this ends up being a lot cleaner.

Every "negation" determines if the node should be `assertIsNone` vs `assertIsNotNone`. I can just increment negations, knowing that 

1) assertFalse adds a negation
2) a UnaryOperation (that is, `not x is None`) adds a negation
3) using `is not` adds a negation

Reply:
This is really neat!